### ISSUE TITLE
Many jquery functions not working while loading web application url

### ISSUE THREAD

**Issue by Faheemmcfc** [electron/electron#11082]
I am trying to load my web application to electron. But many of my application functions are not working in electron. Major problem is dropdown lists are not showing up. I tried loading with window.loadurl and also using webview in index.html.

* Electron version: 1.7.9
* Operating system: Windows 10

### Expected behavior

I like to get my web application with full functionalities working in electron
### Actual behavior

Many jquery functions not working. Dropdowns not showing up...
### How to reproduce

 My current html is: 
```<!DOCTYPE html>
<html>
  <head>
    <title>jQuery injection into webview preload</title>
  </head>
  <body style="overflow:hidden;">
    <webview id="webview" preload="./preload.js" src="http://www.mcgeoautomation.com" style="position:absolute;width:100%;height:100%;"></webview>
    <script>
      // need to have a script tag make webview work even if you don't plan to use it...
    </script>
  </body>
</html>
```

Preload.js is:
```window.onload = function() {
    var script = document.createElement("script");
    script.src = "./jquery.min.js";
    document.body.appendChild(script);
};
```

index.js is:
```const electron = require('electron');
const app = electron.app;
var path=require('path');
const BrowserWindow = electron.BrowserWindow;
var mainWindow;

app.on('ready',function(){
	mainWindow = new BrowserWindow({
		width: 1024, 
		height: 768,
		backgroundColor: '#2e2c29',
		show:false,
	});
	
	mainWindow.loadURL(`file://${__dirname}/webView.html`);
	mainWindow.maximize(true);
	mainWindow.once('ready-to-show',()=>{
		mainWindow.show()
	})

	mainWindow.on('close', (e)=>{
			
		app.quit();
	});
});
```

Package json: 
```{
  "name": "crushmate",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "electron ."
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "electron": "^1.7.9",
    "jquery": "^3.2.1"
  },
  "devDependencies": {
    "electron-prebuilt": "^1.4.13"
  }
}
```

Please help....
Regards

**Comment by MarshallOfSound** [comment]
It looks like you didn't fill out our [Issue Template](https://raw.githubusercontent.com/electron/electron/master/.github/ISSUE_TEMPLATE.md), please ensure you provide all information requested in the issue template so that we can try to help you out.

In the future issues you create without the template will be closed, you can find the issue template [here](https://raw.githubusercontent.com/electron/electron/master/.github/ISSUE_TEMPLATE.md) and it is automatically populated in the new issue form.

**Comment by robinwassen** [comment]
The problem is how jQuery is designed, if `window.module` is defined it expects you to use jQuery by using `require()` which probably is not done on your website.

Your preload script just adds an inclusion of jQuery again, but jQuery will still find that window.module is defined and not bind itself to the window. So there is no difference.

Try to add the following line to the end of your preload.js function:

`window.$ = window.jQuery = require('jquery');`

There is a related SO thread about this issue too: https://stackoverflow.com/questions/32621988/electron-jquery-is-not-defined

**Comment by Faheemmcfc** [comment]
@robinwassen tried your method but still no change.

**Comment by robinwassen** [comment]
@Faheemmcfc I tested to embed your site in a `<webview>` and cannot find any issues with it (at least at the login screen), jQuery seems to load ok and the only error that occurs is a font file not being found.

Modified the https://github.com/electron/electron-quick-start with the following code to test it out.

**Code:**

    <webview id="foo" src="http://www.mcgeoautomation.com" style="display:inline-flex; width:900px; height:600px"></webview>

    <script>
      setTimeout(function() {
        document.getElementById("foo").openDevTools();
      }, 2000);
    </script>

Your best path forward is probably just to fire up the developer tools for the webview and try to figure out what resources that fail to load or whatever it is causing the issue. You will need to provide an error message or something for other to be able to help you further.

**Screenshot:**

![jquery-issue](https://user-images.githubusercontent.com/302151/32698115-8190af48-c79f-11e7-8ea9-c7b892e0b897.PNG)

**Comment by Faheemmcfc** [comment]
@robinwassen There is no problem in login page... Dropdowns are working now. But still problems with saving data in different forms.. Please check by login in with 
username: xyz
password: 123
Check User creation page inside Dashboard. I cannot make a new user in it. Checked the same in browser and it is working... No console errors are seen in it. Please help...

**Comment by robinwassen** [comment]
@Faheemmcfc I checked and I am able to reproduce the issue, it seems to be some problem with `<datalist>`.

Tested to execute the following code on Chrome and Electron on the user creation page: `$('#ddlUserType').val()`. Chrome returned the selected User Type while Electron returned empty string.

There is an open issue for this: https://github.com/electron/electron/issues/9860

Work around would be to use `<select>` instead of `<datalist>`

**Comment by robinwassen** [comment]
A side note to the Electron bug is that every click in your application generates an error (both in Electron and Chrome).

The reason is that your code that dismiss notifications (?) does not null check and tries to run `close()` on a null object.

**The affected code:**
```
    $(".wrapper").click(function (e) {
        if (e.target.id != "btnSubmit") {
                noty.close();
        } 
    });
```

**Error generated:**

```
Uncaught TypeError: Cannot read property 'close' of undefined
    at HTMLDivElement.<anonymous> (Masterpage.js:44)
    at HTMLDivElement.dispatch (jquery-3.1.1.min.js:3)
    at HTMLDivElement.q.handle (jquery-3.1.1.min.js:3)
```

**Comment by robinwassen** [comment]
This issue can be closed as duplicate of #9860
